model/fieldtypes/StringField.php

```diff
@@ -83,7 +83,9 @@ public function getNullifyEmpty() {
 	 * @see core/model/fieldtypes/DBField#exists()
 	 */
 	public function exists() {
-		return ($this->value || $this->value == '0') || ( !$this->nullifyEmpty && $this->value === '');
+		return $this->value
+			|| (is_string($this->value) && strlen($this->value))
+			|| (!$this->nullifyEmpty && $this->value === '');
```

Comment:
Yes, that would be nicer. But still, can we use the getter function `$this->getNullifyEmpty()` rather than directly accessing the property?